File: ts/lang/otterop/lang/ArrayIterator.ts
import { OOPIterator } from './OOPIterator';
import { Array } from './Array';

/** @internal */
export class ArrayIterator<T> implements OOPIterator<T> {

    #array : Array<T>;

    #i : number;

    /** @internal */
    constructor(array : Array<T>) {
        this.#array = array;
        this.#i = 0;
    }

    public hasNext() : boolean {
        return this.#i < this.#array.size();
    }

    public next() : T {
        let ret : T = this.#array.get(this.#i);
        this.#i++;
        return ret;
    }
}

File: python/lang/otterop/lang/_array_iterator.py

class ArrayIterator:
        
    def __init__(self, array):
        self._array = array
        self._i = 0

    def has_next(self):
        return self._i < self._array.size()

    def next(self):
        ret = self._array.get(self._i)
        self._i += 1
        return ret
